File: 703465-delete-smaller-than-next-GFG/703465-delete-smaller-than-next.cpp
class Solution {
	public:
	vector<int> deleteElement(vector<int>& arr, int k) {
		// Code Here
		vector<int> st;
		
		for (int x : arr) {
			while (!st.empty() && k > 0 && st.back() < x) {
				st.pop_back();
				k--;
			}
			st.push_back(x);
		}
		
		return st;
	}
};


// Synced seamlessly with LeetHub Pro
// Pro features: https://bit.ly/leethubpro | Free version: https://bit.ly/leethubv4
// Get it here: https://chromewebstore.google.com/detail/bcilpkkbokcopmabingnndookdogmbna

File: 703465-delete-smaller-than-next-GFG/703465-delete-smaller-than-next.java
class Solution {
	public static ArrayList<Integer> deleteElement(int arr[], int k) {
		// Code here
		ArrayList<Integer> st = new ArrayList<>();
		
		for (int num : arr) {
			while (!st.isEmpty() && k > 0 && st.get(st.size() - 1) < num) {
				st.remove(st.size() - 1);
				k--;
			}
			st.add(num);
		}
		
		return st;
	}
}


// Synced seamlessly with LeetHub Pro
// Pro features: https://bit.ly/leethubpro | Free version: https://bit.ly/leethubv4
// Get it here: https://chromewebstore.google.com/detail/bcilpkkbokcopmabingnndookdogmbna